kernel/init.c

```diff
@@ -661,6 +661,9 @@ FUNC_NORETURN void z_cstart(void)
 	/* perform basic hardware initialization */
 	z_sys_init_run_level(INIT_LEVEL_PRE_KERNEL_1);
 	z_sys_init_run_level(INIT_LEVEL_PRE_KERNEL_2);
+#if defined(CONFIG_SMP)
+	arch_smp_init();
```

Comment:
Is there a reason for calling it in other place (earlier)? We call INIT_LEVEL_SMP here: https://github.com/zephyrproject-rtos/zephyr/blob/593dfe13856df4d2919874037cbb5691b9f00a15/kernel/init.c#L451

Reply:
Then maybe we want to move this into soc/arch layers and out of arch_interface.h entirely?  Really there's zero kernel involvement to the function, it's purely there so the platform layers can do their setup, but when/how isn't really something the OS can (or wants to) decide on.

Reply:
now on arc,  `arch_smp_init` is called in PRE_KERNEL_1., so I am not following.

INIT_LEVEL_SMP has been something we wanted to get rid of, if I am not mistaken, we just have 1-2 users of that level.

Reply:
SMP level and actually initializing smp is indeed completely different.